tangos/properties/pynbody/zoom.py

```diff
@@ -1,12 +1,25 @@
 from . import PynbodyPropertyCalculation
 
+
 class Contamination(PynbodyPropertyCalculation):
+    """ Calculates the contamination of a zoomed halo by heavier, unzoomed particles.
+        The current behaviour returns:
+            1.0 if the halo contains only heavy particles
+            X.X if the halo contains a mixture of heavy and light particles
+            0.0 if the halo contains exclusively light particles
+        Heavies are defined as any particle heavier than the deepest zoom level if multiple zooms are performed
+    """
 
-    def calculate(self, halo, exist):
-        n_heavy = (halo.dm['mass'] > self.min_dm_mass).sum()
-        return float(n_heavy) / len(halo.dm)
+    names = "contamination_fraction"
 
-    def preloop(self, sim, db_timestep):
-        self.min_dm_mass = sim.dm['mass'].min()
+    def calculate(self, halo, exist):
+        loaded_data_min_dm_mass = halo.dm['mass'].min()
+        simulation_min_dm_mass = self._simulation.get("approx_resolution_Msol")
```

Comment:
I think it would be good to add a default (infinity?) value here so that old databases can still calculate the new `contamination_fraction`